### PR TITLE
React/pt 70/tweets list oauth integration

### PR DESCRIPTION
**Pull request by eduardb**
TL;DR Use real data for the tweets list and tweet details

Previously, the authorization header for the aforementioned calls was hardcoded. So now we use the `OauthHelper` to compute the real auhorization header. To achieve this, we defined methods in the `TwitterRequestsService` to help us for that, and added support for query parameters in `OauthHelper`. We also inject the `TwitterRequestsService` across the different views so we don't have to instantiate each time a new service, and we encapsulated the `AuthenticationService` logic in the service. Plus misc refactors to improve quality.

No actual screenshots, as we only touched the functionality, not the UI.

Paired with @gbasile

### CODE REVIEW

File: ReactNative/ReactTwitter/core/views/tweetsList.js [hunk 1]
```diff
@@ -23,23 +25,14 @@ var TweetsList = React.createClass({
...
-      this.setState({
-        dataSource: this.state.dataSource.cloneWithRows(rjson)
+    this.props.twitterService.getHomeTimeline()
+      .then((rjson) => {
+        this.setState({
+          dataSource: this.state.dataSource.cloneWithRows(rjson)
+        })
       })
-    })
+      .catch((err) => { console.warn(err) })
```
Comment (xrigau) [inline ReactNative/ReactTwitter/core/views/tweetsList.js]: 👍

**Comment by dominicfreeston** [comment]
I've looked at this multiple times now but I lack too much context to provide any useful feedback, but 👍 from me FWIW.

File: ReactNative/ReactTwitter/core/views/login-screen.js [hunk 2]
```diff
@@ -11,18 +11,16 @@ var Button = require('react-native-button')
 var TwitterRequestsService = require('../service/twitter-requests-service.js')
 var DeepLinkingFacade = require('../service/deep-linking-facade')
 var OauthHelper = require('../service/oauth-helper.js')
-var AuthenticationService = require('../service/authentication-service.js')
 
 var LoginScreenView = React.createClass({
   propTypes: {
-    navigator: React.PropTypes.instanceOf(Navigator).isRequired
+    navigator: React.PropTypes.instanceOf(Navigator).isRequired,
+    twitterService: React.PropTypes.instanceOf(TwitterRequestsService).isRequired
```
Comment (dominicfreeston) [inline ReactNative/ReactTwitter/core/views/login-screen.js]: [Question] Does `isRequired` set a constraint, or does it actually create the instance here?

Comment (eduardb) [inline ReactNative/ReactTwitter/core/views/login-screen.js]: It's just a constraint for validation, the instance is sent using the component's `props` from here: https://github.com/novoda/spikes/blob/react/PT-70/tweets-list-oauth-integration/ReactNative/ReactTwitter/core/views/main-navigator.js#L37

**Comment by xrigau** [comment]
:shipit:
